maven-embedder/src/main/java/org/apache/maven/cli/internal/ExtensionResolutionException.java

```diff
@@ -0,0 +1,59 @@
+package org.apache.maven.cli.internal;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.cli.internal.extension.model.CoreExtension;
+
+/**
+ * Exception occurring trying to resolve a plugin.
+ *
+ * @author <a href="mailto:brett@apache.org">Brett Porter</a>
+ */
+public class ExtensionResolutionException
+    extends Exception
+{
+
+    private final CoreExtension extension;
+
+    public ExtensionResolutionException( CoreExtension extension, Throwable cause )
+    {
+        super( "Extension " + getId( extension ) + " or one of its dependencies could not be resolved: "
+                        + cause.getMessage(), cause );
```

Comment:
I guess this duplicated the previous exception, but duplicating the cause message isn't right.

Reply:
Right, but if the interface is enhanced, the exception will be moved along the new method I suppose.

Reply:
Agreed, make it wrong, but consistent with the rest.